### PR TITLE
Fix duplicate entries in HEADER_SEARCH_PATHS when running react_native_post_install script

### PR DESCRIPTION
**Pull request by dirkpostma**
## Summary:

In a react native project where USE_FRAMEWORKS is not nil, every time when running `pod install`, duplicate lines are added to `HEADER_SEARCH_PATHS` section of `project.pbxproj`:

```
" ${PODS_CONFIGURATION_BUILD_DIR}/ReactCommon/ReactCommon.framework/Headers",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-Fabric/React_Fabric.framework/Headers/react/renderer/components/view/platform/cxx",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-graphics/React_graphics.framework/Headers",
" ${PODS_CONFIGURATION_BUILD_DIR}/ReactCommon/ReactCommon.framework/Headers",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-Fabric/React_Fabric.framework/Headers/react/renderer/components/view/platform/cxx",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-graphics/React_graphics.framework/Headers",
" ${PODS_CONFIGURATION_BUILD_DIR}/ReactCommon/ReactCommon.framework/Headers",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-Fabric/React_Fabric.framework/Headers/react/renderer/components/view/platform/cxx",
" ${PODS_CONFIGURATION_BUILD_DIR}/React-graphics/React_graphics.framework/Headers",
```

Note: a popular library that needs `use_frameworks` is react-native-firebase.
See https://rnfirebase.io/#altering-cocoapods-to-use-frameworks

## Analyse

- `react_native_post_install` calls `ReactNativePodsUtils.update_search_paths(installer)`
- when `ENV['USE_FRAMEWORKS'] != nil` then `update_search_paths` calls `add_search_path_if_not_included`
- `add_search_path_if_not_included` checks if `"#{new_search_path}"` is already there
- if not found it adds `" #{new_search_path}"` _with an extra space_
- next time, it can't find `"#{new_search_path}"` because of the extra space, and adds `" #{new_search_path}"` again

## Changelog:

[IOS] [FIXED] - react_native_post_install script no longer adds duplicate entries to HEADER_SEARCH_PATHS

## Test Plan:

- create a react native project
- add `use_frameworks! :linkage => :static` to `ios/Podfile` (just before `use_react_native`)
- run `pod install`
- assert no duplicate lines are added to HEADER_SEARCH_PATHS of file `project.pbxproj`

### CODE REVIEW

**Comment by cipolleschi** [comment]
/rebase - this comment automatically rebase on top of main

File: packages/react-native/scripts/cocoapods/utils.rb [hunk 1]
```diff
@@ -539,7 +539,7 @@ def self.remove_flag_for_key(config, flag, key)
 
     def self.add_search_path_if_not_included(current_search_paths, new_search_path)
         if !current_search_paths.include?(new_search_path)
```
Comment (cipolleschi) [inline packages/react-native/scripts/cocoapods/utils.rb:541]:
```suggestion
        entry = " #{new_search_path}"
        if !current_search_paths.include?(new_search_path) && !current_search_paths.include?(entry)
          current_search_paths << entry
```

**Review by cipolleschi — CHANGES_REQUESTED** [review]

File: packages/react-native/scripts/cocoapods/utils.rb (hunk 1 above)
Comment (dirkpostma) [reply packages/react-native/scripts/cocoapods/utils.rb]: @cipolleschi Thanks for pointing this out! I didn't realise the input could be both string or array. 

I updated the code a bit different than your suggestion. I didn't really like the solution where we depend on extra spaces to be added. In stead, I now strip them if they exist, resulting in a cleaner project without unnecessary spaces.

Do you think it's okay like this? Or is there are reason to keep the "extra spaces" solution?

I also added some tests for the function.

You can run them using by running: `ruby packages/react-native/scripts/cocoapods/__tests__/utils-test.rb -n "/add_search_path_if_not_included/" --verbose`

Results:

```
Loaded suite packages/react-native/scripts/cocoapods/__tests__/utils-test
Started
UtilsTests:
  test_add_search_path_if_not_included_with_array_input:								.: (0.000148)
  test_add_search_path_if_not_included_with_invalid_input:								.: (0.000059)
  test_add_search_path_if_not_included_with_string_input:								.: (0.000040)

Finished in 0.00047 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------
3 tests, 7 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------
6382.98 tests/s, 14893.62 assertions/s
```

**Comment by dirkpostma** [comment]
I thought of one possible issue: folder names with spaces. Original code preserves these, my code doesn't. Not sure if this ever is a problem, but to be sure, although we keep the the extra spaces, I think your suggestion is better.

File: packages/react-native/scripts/cocoapods/utils.rb (hunk 1 above)
Comment (dirkpostma) [reply packages/react-native/scripts/cocoapods/utils.rb]: I thought of a possible issue: folder names with spaces. Original code preserves these, my code did't. Updated function and tests to support folders with spaces.

**Review by cipolleschi — APPROVED** [review]
Thanks for doing this! It's amazing that you added also the tests!
That's the best solution: I didn't want to suggest it in advance as it was quite a bit of work that I would have put on your shoulders, but I'm glad you did it!

**Comment by dirkpostma** [comment]
@cipolleschi I a lint build
Step failed on the meta internal CI. Unfortunately I can’t view the logs there and I don’t know the minting rules.. 

I did make some changes with regard to whitespacing in the last commit, but that’s a bit of a guess. 

If this doesn’t fix it, can you do it? Thanks!

**Pull request closed without merging**